src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterOfT.cs

```diff
@@ -216,6 +216,20 @@ internal bool TryRead(ref Utf8JsonReader reader, Type typeToConvert, JsonSeriali
             bool isContinuation = state.IsContinuation;
             bool success;
 
+            if (
+#if NETCOREAPP
+                !typeof(T).IsValueType &&
+#endif
```

Comment:
Why only for netcoreapp?  I understand typeof(T).IsValueType is a JIT-time constant on recent .NET Cores, but if it's cheaper than the expensive push/pops this is trying to avoid, it's not still valuable even when it's not a constant?

Reply:
It is implied by `CanBePolymorphic` so it would be redundant as a runtime check.

Reply:
Ok